BE/src/main/java/team29/hoorry/issuetracker/core/label/LabelService.java

```diff
@@ -0,0 +1,51 @@
+package team29.hoorry.issuetracker.core.label;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import team29.hoorry.issuetracker.core.label.domain.Label;
+import team29.hoorry.issuetracker.core.label.dto.LabelAddRequest;
+import team29.hoorry.issuetracker.core.label.dto.LabelResponse;
+import team29.hoorry.issuetracker.core.label.dto.LabelUpdateRequest;
+import team29.hoorry.issuetracker.core.label.dto.LabelsResponse;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class LabelService {
+
+	private final LabelRepository labelRepository;
+
+	public LabelsResponse findAll() {
+		List<LabelResponse> labelResponses = labelRepository.findAll().stream()
+			.map(LabelResponse::from)
+			.collect(Collectors.toList());
+
+		return new LabelsResponse(labelResponses);
+	}
+
+	public void save(LabelAddRequest labelAddRequest) {
+		Label newLabel = labelAddRequest.toEntity();
+		labelRepository.save(newLabel);
+	}
+
+	public void update(Long id, LabelUpdateRequest labelUpdateRequest) {
+		Label label = labelRepository.findById(id)
+			.orElseThrow(() -> new NoSuchElementException("존재하지 않는 labelId입니다."));
+		label.update(
+			labelUpdateRequest.getTitle(),
+			labelUpdateRequest.getTitleColor(),
+			labelUpdateRequest.getBackgroundColor(),
+			labelUpdateRequest.getDescription()
+		);
+	}
+
+	public void delete(Long id) {
+		Label label = labelRepository.findById(id)
+			.orElseThrow(() -> new NoSuchElementException("존재하지 않는 labelId입니다."));
+		labelRepository.delete(label);
```

Comment:
```suggestion
		try {
			labelRepository.deleteById(id);
		} catch (EmptyResultDataAccessException e) {
			throw new NoSuchElementException("존재하지 않는 labelId입니다.");
		}
```

모가 더 낫다고 생각하시나요,,, 저는 후자가 너무 못생겨서 findById한 후에 delte(Entity entity)사용하였습니다,,,

Reply:
![스크린샷 2022-06-27 오전 3 28 28](https://user-images.githubusercontent.com/68011320/175830259-691376e3-04c6-4761-ab21-2d5cf0db1935.png)

+) springdatajpa가 제공하는 delete, deleteById 등등 id가 존재하지 않을 때 예외를 던져주기 위해서 select 쿼리가 같이 나가고 있습니다. 저는 delete 요청의 경우, 존재하는 id나 존재하지 않는 id나 수행결과가 똑같으니까 select문이 불필요하다고 생각하는데 어떻게 생각하시나요? 이런 경우에는 따로 jpql이나 querydsl을 이용하여 쿼리를 따로 짜줘야할까요?

Reply:
두 번 DB를 가는 것보다 직접 jpql로 작성하는 것이 좋을 것 같아요!